daemon/exec.go

```diff
@@ -206,8 +194,8 @@ func (d *Daemon) ContainerExecStart(name string, stdin io.ReadCloser, stdout io.
 		}
 
 		// Maybe the container stopped while we were trying to exec
-		if !c.IsRunning() {
-			return derr.ErrorCodeExecContainerStopped
+		if !c.IsRunningLocking() {
```

Comment:
Self comment:
`exec` can only work on running container, exclude `paused` and `restarting`